### ISSUE TITLE
Tie in select tags to SPA page switching

### ISSUE THREAD

**Issue by bob2517-whiteline** [Active-CSS/active-css#314]
Got a project at the moment where the dropdown switches pages. Not the first time I've had to do that, but it's the first time for an SPA. Using an ACSS location command on the {@@value} when using an input event doesn't fire off the SPA handling. It does switch the page ok, but it doesn't stay in the SPA, which just doesn't cut the cheese for me.

As page switchers are a common scenario, there could be a method of triggering the SPA page switch automatically without additional ACSS coding, by putting the page name in a custom attribute in the option tag, like data-page. Would be better to do this than to use the value attribute, which has to remain primarily a form control. Usually doing stuff like this, I would put the page route into the value attribute and use a onchange location switch on that value, but 1) it wouldn't be a performant choice to analyse the option value for SPA page existence on every change, and 2) that value actually may be needed for form submittance and not a page switch and default behaviour shouldn't be overridden. Better off having a dedicated attribute for it.

So for example:
```
<select id="branchChooser">
<option>Choose a branch</option>
<option data-page="/aylesford">Aylesford</option>
<option data-page="/brighton">Brighton</option>
<option data-page="/eastbourne">Eastbourne</option>
<option data-page="/hastings">Hastings</option>
<option data-page="/orpington">Orpington</option>
<option data-page="/tunbridge-wells">Tunbridge Wells</option>
</select>
```

Selecting an option with a data-page attribute that matches a route from @pages would act like an "a" tag without any ACSS coding needed. That's the theory anyway - need to work on it to see if it will work.

**Comment by bob2517** [comment]
This is now on the branch. The syntax is as above. When the dropdown option is selected, the SPA rules for grabbing the page href from @pages kicks in and that data comes from the data-pages attribute. The route attributes from @pages are then placed by the core into the select tag, which can then be used as a reference later, and for determining what to do with dropdown change. The select tag then gets the change event fired on itself by the core which then triggers the change in the config if it is present. The change event, therefore, needs to be set up in the config alongside the click event that is normally used for SPA page switching.

ie.:
```
.ajaxLink:click, .ajaxLink:change {
    ... do the ajax loading, rendering, etc.
}
```

So in summary, all that is needed to get a dropdown triggering a route change is to use a data-page attribute pointing to the route in the option tags, and then set up a change event to determine what happens when the option is selected.

This is now on the branch. It works in both Firefox and chromium.

**Comment by bob2517** [comment]
Closing pending release.
